### ISSUE TITLE
Update of resource `backupdr_template` fails with `Could not update template, unexpected error: 405 Method Not Allowed`

### ISSUE THREAD

**Issue by daniel-cit** [umeshkumhar/terraform-provider-backupdr#2]
**TL;DR**
Updating a `backupdr_template` resource  always fails with `Could not update template, unexpected error: 405 Method Not Allowed`

**Details**
Terraform configuration used

```hcl
resource "backupdr_template" "no_policies" {
  name        = "bkdrtestemptyreally"
  description = "SLA Template test no policies"
}
```

the template is created successfully after `terraform apply`

doing a simple change

```hcl
resource "backupdr_template" "no_policies" {
  name        = "bkdrtestemptyreally"
  description = "SLA Template test no policies - edited"
}
```

and doing a `terraform apply` fails with

```bash
2024-06-18T21:18:01.718-0300 [ERROR] vertex "backupdr_template.no_policies" error: Error Updating SLA Template
╷
│ Error: Error Updating SLA Template
│ 
│   with backupdr_template.no_policies,
│   on backup.tf line 130, in resource "backupdr_template" "no_policies":
│  130: resource "backupdr_template" "no_policies" {
│ 
│ Could not update template, unexpected error: 405 Method Not Allowed
```

If we try to do the same update using `curl` it works

```bash
curl -X PUT https://bmc-REDACTED-REDACTED-dot-us-central1.backupdr.googleusercontent.com/actifio/slt/7348 \
    --header "Authorization: Bearer $(gcloud auth print-access-token)" \
    --header "backupdr-management-session: Actifio REDACTED" \
    --header 'Accept: application/json' \
    --header "Content-Type: application/json" \
    -d '{ "description" : "SLA Template test no policies - edited", "name" : "bkdrtestemptyreally"}'
```

response
```
{
   "@type" : "sltRest",
   "id" : "7348",
   "href" : "http://bmc-REDACTED-REDACTED-dot-us-central1.backupdr.googleusercontent.com/actifio/slt/7348",
   "description" : "SLA Template test no policies - edited",
   "name" : "bkdrtestemptyreally",
   "override" : "false",
   "policy_href" : "http://bmc-REDACTED-REDACTED-dot-us-central1.backupdr.googleusercontent.com/actifio/slt/7348/policy"
}
```

adding a `Printf` to debug the Update call
https://github.com/umeshkumhar/terraform-provider-backupdr/blob/main/internal/provider/template_resource.go#L371C2-L378C3
```
	respObject, res, err := r.client.SLATemplateApi.UpdateSlt(r.authCtx, plan.ID.ValueString(), &reqBody)
	if err != nil {
		log.Printf("\n\n [DEBUG] ID=%s reqBody=%+v \n\n", plan.ID.ValueString(), &reqBody)
		resp.Diagnostics.AddError(
			"Error Updating SLA Template",
			"Could not update template, unexpected error: "+err.Error(),
		)
		return
	}
```  

we got
```
2024-06-18T21:18:01.710-0300 [DEBUG] provider.terraform-provider-backupdr:  [DEBUG] ID= reqBody=&{Body:{isSet:true value:{Managedby: Resource: Metadata:<nil> PolicyHref: OptionHref: Source:[] Description:SLA Template test no policies - edited Override: Sourcename: Policies:[] Options:[] Orglist:[] Managedbyagm:false Usedbycloudapp:false Name:bkdrtestemptyreally Href: Syncdate:0 Stale:false Id:}}}
```
showing that the value for `plan.ID.ValueString()` is the empty string `""`

the `plan.ID.ValueString()` is the ID of the template as in the definition of the `func`
```
func (*backupdr.SLATemplateApiService).UpdateSlt(ctx context.Context, sltId string, localVarOptionals *backupdr.SLATemplateApiUpdateSltOpts) (backupdr.SltRest, *http.Response, error)
```

if we hardcode the correct ID in the `UpdateSlt` call to do a quick test
```
respObject, res, err := r.client.SLATemplateApi.UpdateSlt(r.authCtx, "7348", &reqBody) // plan.ID.ValueString()
```
The update works

debugging the plan variable at https://github.com/umeshkumhar/terraform-provider-backupdr/blob/main/internal/provider/template_resource.go#L322
```
func (r *templateResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
	// Retrieve values from plan
	var plan templateResourceModel
	diags := req.Plan.Get(ctx, &plan)
	log.Printf("\n\n\n\n [DEBUG] plan=%+v \n\n\n\n", &plan)
```

we have
```
2024-06-18T22:18:17.658-0300 [DEBUG] provider.terraform-provider-backupdr:  [DEBUG] plan=&{ID:<unknown> Href:<unknown> Name:"bkdrtestemptyreally" Description:"SLA Template test no policies - really - edited - once more" OptionHref:<unknown> PolicyHref:<unknown> Sourcename:<null> Override:<null> Policies:[] Managedbyagm:<null> Usedbycloudapp:<null> Syncdate:<null> Stale:<null>} 
```
which looks like the cause of the missing ID in the update call

This may also explain why the resources do not identify when a resource is changed outside of Terraform 

[Writing log output](https://developer.hashicorp.com/terraform/plugin/log/writing) in the repository code to log the parameters of the APIs calls at `DEBUG` level would be helpful in other investigations.
